### PR TITLE
[PPL] Added integration tests for query analysis and search, where, fields commands

### PR DESCRIPTION
**Pull request by chloe-zh**
*Issue #, if available:*

*Description of changes:*

- Changed the default value of `argList` in `Field` from `null` to empty list to avoid `NullPointerException`, and added `hasArgument()` method to `Field`, `Project` and `Aggregation` nodes.
- Added integration tests for query analysis and search, where, fields commands

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        String query = String.format("search source=%s | fields name", TEST_INDEX_ACCOUNT);
+        queryShouldThrowSemanticException(query, "can't resolve expression name in type env");
+    }
+
+
+    private void queryShouldPassSemanticCheck(String query) {
+        Request request = buildRequest(query);
+        try {
+            Response response = client().performRequest(request);
+            assertEquals(OK.getStatus(), response.getStatusLine().getStatusCode());
```
Comment (dai-chen) [inline integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java:108]: This is same as `executeQuery`? Probably we can just call it and discard return value?

```
    protected String executeQuery(String query) throws IOException {
        Response response = client().performRequest(buildRequest(query));
        Assert.assertEquals(200, response.getStatusLine().getStatusCode());
        return getResponseBody(response, true);
    }
```

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        queryShouldPassSemanticCheck(query);
+    }
+
+
+    /**
+     * Commands that fail semantic analysis should throw {@link SemanticCheckException}
+     */
+    @Test
+    public void unsupportedAggregationShouldFailSemanticCheck() {
+        String query = String.format("search source=%s | stats count(age)", TEST_INDEX_ACCOUNT);
```
Comment (dai-chen) [inline integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java:93]: Is count being added in https://github.com/opendistro-for-elasticsearch/sql/pull/493?

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        queryShouldThrowSemanticException(query, "Unsupported aggregation function count");
+    }
+
+    @Test
+    public void nonexistentIndexShouldFailSemanticCheck() {
+        String query = String.format("search source=%s | fields name", TEST_INDEX_ACCOUNT);
```
Comment (dai-chen) [inline integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java:99]: The test method name should be nonexistent field? But I think we do need another one for index.

Comment (chloe-zh) [reply integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java:99]: It should be nonexistent field, corrected it now. It seems we have not set semantic check for index, let's add the index check when the index check is enabled.

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java (hunk 2 above)
Comment (chloe-zh) [reply integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java:93]: Thanks for the reminder. Changed the case to `range`, is that making sense?

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java (hunk 1 above)
Comment (chloe-zh) [reply integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java:108]: Done, thanks!

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java (hunk 3 above)
Comment (dai-chen) [reply integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java]: The `ElasticsearchIndex.getFieldTypes()` should throw Elasticsearch IndexNotFoundException. Not sure if the exception is caught without re-throw or for other reason.

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java (hunk 2 above)
Comment (dai-chen) [reply integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java]: Sure. Thanks!

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java (hunk 3 above)
Comment (chloe-zh) [reply integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java]: Yes Elasticsearch throws this exception. I didn't add it here because it's not handled by the semantic check. Maybe we can add this to other test

Comment (dai-chen) [reply integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java]: Sounds good!

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/SearchCommandIT.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Override
+    public void init() throws IOException {
+        loadIndex(Index.BANK);
+    }
+
+    @Test
+    public void testSearchAllFields() throws IOException {
+        String result = executeQuery(String.format("search source=%s", TEST_INDEX_BANK));
+        assertTrue(result.contains("\"total\": 7"));
```
Comment (penghuo) [inline integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/SearchCommandIT.java:33]: Is it possible to assert all the response instead of just one field. e.g.
schema(contains("name", "string")))
or we can assert the full string match.
Actually, we don't need much data for integration test purpose, IMO, the sample data include all the types, null, missing should be enough for IT purpose.

Comment (chloe-zh) [reply integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/SearchCommandIT.java:33]: I'm considering to add a utils to build result set and schema from the result, so that we can test the behavior in a more accurate way.
For the sample data, what about applying `bank_with_null_values` (already in our resources) to all tests?
